test/e2e/baseFixture.ts

```diff
@@ -1,12 +1,71 @@
+import { field, logger } from "@coder/logger"
 import { test as base } from "@playwright/test"
-import { CodeServer } from "./models/CodeServer"
+import { CodeServer, CodeServerPage } from "./models/CodeServer"
 
-export const test = base.extend<{ codeServerPage: CodeServer }>({
-  codeServerPage: async ({ page }, use) => {
-    const codeServer = new CodeServer(page)
-    await codeServer.navigate()
-    await use(codeServer)
+/**
+ * Wraps `test.describe` to create and manage an instance of code-server. If you
+ * don't use this you will need to create your own code-server instance and pass
+ * it to `test.use`.
+ *
+ * If `includeCredentials` is `true` page requests will be authenticated.
+ */
+export const describe = (name: string, includeCredentials: boolean, fn: (codeServer: CodeServer) => void) => {
+  test.describe(name, () => {
+    // This will spawn on demand so nothing is necessary on before.
+    const codeServer = new CodeServer(name)
+
+    // Kill code-server after the suite has ended. This may happen even without
+    // doing it explicitly but it seems prudent to be sure.
+    test.afterAll(() => {
+      codeServer.close()
```

Comment:
Do we need to `await` this since `close()` is async?
```suggestion
    test.afterAll(async () => {
      await codeServer.close()
```

Reply:
Nice catch!